vstools/utils/vs_proxy.py

```diff
@@ -169,10 +169,12 @@ def clear_cache() -> None:
         try:
             for output in get_outputs().values():
                 if isinstance(output, VideoOutputTuple):
-                    output.clip.get_frame(0)
+                    with output.clip.get_frame(0):
+                        pass
```

Comment:
same `try`/`finally` idea?

Reply:
Same thing, though I found out this is apparantly valid
```py
                if isinstance(output, VideoOutputTuple):
                    output.clip.get_frame(0).close()
                    break
```

Reply:
There’s nothing between the get_frame call and the close call, so no opportunities for other exceptions. If get_frame throws, there’s nothing to be closed. If it succeeds, you immediately close it anyway.

Reply:
Sounds good, `get_frame().close()` should be perfectly fine then

Reply:
Ironically, I believe this is actually _safer_ than a `with` or a `try`–`finally` with an assignment if the code is running in the main thread of command-line `python`, because those constructs have extra bytecode ops (for the `__exit__` call, the `frame=` assignment, etc.), each of which is an additional opportunity for a `KeyboardInterrupt` to occur.